src/main/scala/com/codacy/plugins/api/results/Pattern.scala

```diff
@@ -30,26 +30,23 @@ object Pattern {
 
   case class Specification(patternId: Pattern.Id,
                            level: Result.Level,
-                           category: Specification.Category,
-                           subcategory: Option[Specification.Subcategory],
+                           category: Category,
+                           subcategory: Option[Subcategory],
                            parameters: Option[Set[Parameter.Specification]],
                            languages: Option[Set[Language]] = None) {
-    require(subcategory.isEmpty || category == Specification.Category.Security,
-            "Security is the only category having subcategories")
+    require(subcategory.isEmpty || category == Category.Security, "Security is the only category having subcategories")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of untyped comparison operator](https://app.codacy.com/gh/codacy/codacy-plugins-api/pullRequest?prid=4820049)